rs/crypto/secp256k1/tests/tests.rs

```diff
@@ -146,6 +146,39 @@ fn should_accept_ecdsa_signatures_that_we_generate() {
     }
 }
 
+#[test]
+fn bitcoin_library_accepts_our_bip341_signatures() {
+    use bitcoin::{
+        hashes::hex::FromHex,
+        schnorr::TapTweak,
+        secp256k1::{schnorr::Signature, Message, Secp256k1, XOnlyPublicKey},
+        util::taproot::TapBranchHash,
+    };
+
+    let secp256k1 = Secp256k1::new();
+
+    let mut rng = test_rng();
+
+    for _trial in 0..1024 {
+        let sk = PrivateKey::generate_using_rng(&mut rng);
+
+        let msg = rng.gen::<[u8; 32]>();
+        let ttr = rng.gen::<[u8; 32]>();
+
+        let sig = sk.sign_message_with_bip341(&msg, &mut rng, &ttr).unwrap();
+
+        let pk = XOnlyPublicKey::from_slice(&sk.public_key().serialize_sec1(true)[1..]).unwrap();
+
+        let tnh = TapBranchHash::from_hex(&hex::encode(ttr)).unwrap();
+
+        let dk = pk.tap_tweak(&secp256k1, Some(tnh)).0.to_inner();
```

Comment:
Does `dk` mean derived key? I think the BIPs always speak about this as tweaking. Maybe we could use that instead of "derive" in general.

Reply:
Renamed